### DISCUSSION TITLE
Difference in behavior next/link vs. @/i18n/routing/Link

### DISCUSSION THREAD

**Discussion by marcvives** [amannn/next-intl#1518] [Q&A]
I have a NextJS server function that passes Link as parameter to a MaterialUI component
![image](https://github.com/user-attachments/assets/a2bd07ce-b2a6-49af-a250-26efcc6a0f95)

It worked fine with Next-intl 3.20 , NextJS 14.2.14 and MUI 6.1.2

Now with Next-intl 3.24 , NextJS 15.0.2 and MUI 6.1.6 it works with

![image](https://github.com/user-attachments/assets/2cb246f7-4f31-4ed9-9b00-d62e2c1de52d)
 reverting to the default locale (which makes sense)

but if I import Link from next-intl
![image](https://github.com/user-attachments/assets/8e63d14d-4ca6-4789-9117-a2e6dcf3a296)

I get the following error:

![image](https://github.com/user-attachments/assets/bf8c60e7-b257-4b99-ac10-f1ebb1c11db5)

But next-intl "Link" is supposed to wrap NextJS "link" and localize the pathname as necessary only. Wrapping non-anchor elements and adding an href attribute shouldn't be affected by a package update. Any hints would be welcome.

**Comment by amannn** [comment]
This might help: https://github.com/amannn/next-intl/issues/1271#issuecomment-2331863819

Search terms for future readers:
> Functions cannot be passed directly to Client Components

**Reply by marcvives** [reply]
It works.

You only have to import Link as a client component:

![image](https://github.com/user-attachments/assets/02805347-0c22-407e-bee2-9751da9b4993)

and then 

![image](https://github.com/user-attachments/assets/7f63aa9e-8d83-4990-ab0b-be17d8175749)

Thank you !!
